src/LanguageServer/Impl/Program.cs

```diff
@@ -71,6 +75,19 @@ internal static class Program {
             }
         }
 
+        private static void EnableProfileOptimization() {
+            try {
+                // create directory for profile optimization
+                var path = Path.Combine(Path.GetTempPath(), "PythonLanguageServer", "Profiles");
```

Comment:
another option is creating the profile folder along with .cache folder? but for now, I am creating it at temp directory

Reply:
You get cache folder location on init, not at the startup so I am not sure if it can be used.

Reply:
The cache folder location isn't dependent on any user information, just some specific set of rules that CacheFolder uses. It could be refactored out, since everything needed for it is created in Main.

Alternatively, would it be too harmful to just do this work on initialization, even if it's a little late and weird?

Reply:
putting this profile in temp directory doesn't harm as well. so temp directory shouldn't be an issue as well. I can put user name in path for linux if we care.